docs/reference/modules/piece-page-type.md

```diff
@@ -0,0 +1,240 @@
+  options: {
+    // The previous article piece would be returned with only the `title`, `_url`
+    // and `lastPublishedAt` properties.
+    next: {
+      project: {
+        title: 1,
+        _url: 1,
+        lastPublishedAt: 1
+      }
+    }
+  },
+  // …
+}
+```
+<template v-slot:caption>
+  modules/article-page/index.js
+</template>
+</AposCodeBlock>
+
+## Related Documentation
+
+* [Piece index and show pages](https://v3.docs.apostrophecms.org/guide/piece-pages.html)
+* [Piece page type options](https://v3.docs.apostrophecms.org/reference/module-api/module-options.html#options-for-piece-page-types)
+
+## Featured methods
+The following methods belong to this module and may be useful in project-level code. See the [source code](https://github.com/apostrophecms/apostrophe/blob/main/modules/%40apostrophecms/piece-page-type/index.js) for all the modules that belong to this module.
+
+### `indexQuery(req)`
+This method should be overridden for a piece-type to call additional [query builders](https://v3.docs.apostrophecms.org/reference/query-builders.html#query-builders) by default.
+
+### `beforeIndex(req)`
+This method is called before `indexPage`. It is a convenient method to extend for manipulating the `req` being supplied to that page.
+
+### `beforeShow(req)`
+This method is called before `showPage`. It is a convenient method to extend for manipulating the `req` being supplied to that page.
+
+### `dispatchAll()`
+This method can be extended to override the default behavior of invoking `showPage` if the URL has an additional path after the base, e.g. `/blog/good-article`. As example, you could override to use `/:year/:month/:day/:slug` to invoke `self.showPage`. This should be used in conjunction with the [`buildUrl()`](#buildurl-req-page-piece)) method of this module. See [@apostrophecms/page-type](https://github.com/apostrophecms/apostrophe/blob/main/modules/%40apostrophecms/page-type/index.js) for more about what you can do with dispatch routes.
+
+### `buildUrl(req, page, piece)`
+This method should be extended to build custom URL for use in the [`dispatchAll`](#dispatchall) method.
+
+### `filterByIndexPage(query, page)`
+This method invokes query builders on the supplied query argument to ensure it only fetches results appropriate to the given page. This is typically done when there is more than one pieces-page per page type. This should be used in conjunction with the [`chooseParentPage`](#chooseparentpage-pages-piece) method of this module.
+
+### `chooseParentPage(pages, piece)`
+This `pages` parameter of this method takes an array of all of the index pages for a particular piece-type, and the `piece` parameter is an individual piece-type name. By default, it will return the first item in the `pages` array, but the developer should map pieces to pages in a way that makes sense for their design. This method will give a warning if the [`filterByIndexPage`](#filterbyindexpage-query-page) method has not been overridden.
```

Comment:
This -> The

"The default version of this method will give a warning if it sees more than one page in the array, as it is up to you to override this method to provide a sensible way of deciding which page is the best 'parent' for each piece."

It's not about whether `filterByIndexPage` has been overridden, the warning just mentions that it's a good idea to override that too in the same situation, so each index page displays the same pieces that would logically regard it as a "parent."

Reply:
Changed. I have to think about this one a little more for my understanding.